src/concrete/ml/sklearn/base.py

```diff
@@ -1307,18 +1307,31 @@ def fit(self, X: Data, y: Target, **fit_parameters):
         # Check that the underlying sklearn model has been set and fit
         assert self.sklearn_model is not None, self._sklearn_model_is_not_fitted_error_message()
 
+        self._is_fitted = True
+
+        # Convert the tree inference with Numpy operators
+        # Adjust the auto rounder manually
+        self._convert_tree_to_numpy_and_compute_lsbs_to_remove(q_X, use_rounding=True)
+
+        return self
+
+    def _convert_tree_to_numpy_and_compute_lsbs_to_remove(
```

Comment:
I'm not sure why this method needs to be created ? why not directly call `tree_to_numpy` directly in `fit` as before ?